geode-core/src/main/java/org/apache/geode/internal/net/BufferPool.java

```diff
@@ -394,4 +370,36 @@ public ByteBuffer getBB() {
     }
   }
 
+  public static class PooledByteBuffer {
+    private final ByteBuffer original;
+    private final ByteBuffer byteBuffer;
+
+    PooledByteBuffer(ByteBuffer original, ByteBuffer byteBuffer) {
```

Comment:
Good. I verified the only place this is being called outside the class is in the one place the pool slices a buffer before returning it.